### PR TITLE
Use CircleCI contexts to inject secrets into jobs

### PR DESCRIPTION
**Pull request by owais**
This will ensure only the jobs that require the secrets get access to
them as opposed to all jobs getting access via environment variables.

Contexts can also be shared between projects so we'll have a single
place to manage secrets for both core and contrib CI jobs.

I'll follow up with another change next week that'll further restrict
access to contexts only to maintainers (or a new publishers group).

### CODE REVIEW

**Comment by owais** [comment]
Sure. It's pretty well explained in the official docs here. https://circleci.com/docs/2.0/contexts/#overview

The benefits over the environment variables feature are:

1. Contexts variables are not globally injected into all jobs but must be explicitly enabled for each job that needs them. This means only release job will get access to docker hub access token for example.
2. Contexts allow tighter control over who views or manage the variables. We can restrict a context only to github maintainers for example. This will prevent approvers or triagers from managing the "protected" contexts.
3. Contexts allow jobs to require manual approval. For example, if a release job requires a context that contains docker hub credentials, we can add a manual approval before the release job. If the context is restricted to only maintainers then only maintainers will be able to approve the job. This adds one more additional step to releases (a button click in CircleCI) but ensures only a restricted group can issue releases or access the vars (though some logging or ssh trickery).
 
 
This PR only implements 1. from the above list as a first step. Irrespective of whether we do 2 and 3, 1 is still useful as it prevents tokens from being injected into jobs that don't need them.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
